src/NumericalAlgorithms/Spectral/Basis.cpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Spectral/Basis.hpp"
+
+#include <ostream>
+#include <vector>
+
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/GetOutput.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace Spectral {
+
+std::vector<Basis> all_bases() {
```

Comment:
Switch to std::array? (Unrelated to this PR, just noticed while you're moving the code)